simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,395 @@
+
+        return people
+
+    def __generateKnows(self, people):
+        """
+        Return the knows truth matrix for all people in the dataset.
+        """
+        knowsMatrix = np.zeros((self.numberOfPeople, self.numberOfPeople))
+        for pair in combinations(people, 2):
+            similarity = self.__calculateSimilarity(pair[0], pair[1])
+            knows = self.__flipCoin(self.coinFlipSeed, similarity)
+
+            # Set both (i,j) and (j,i) since P1 knows P2 => P2 knows P1.
+            knowsMatrix[pair[0].index][pair[1].index] = knows
+            knowsMatrix[pair[1].index][pair[0].index] = knows
+
+        return knowsMatrix
+
+    def __writeDataToFile(self, data, paths):
+        """
+        Write data to file.
+        """
+        if self.outputDir:
+            if self.outputDir[0] == '~':
+                prefix = os.path.expanduser(self.outputDir)
+            else:
+                prefix = self.outputDir
+        else:
+            prefix = ''
+
+        for key in data:
+            # We have the same keys for data and its corresponding path.
+            with open(os.path.join(prefix, paths[key]), 'w') as f:
+                if key == 'config':
+                    json.dump(data[key], f, indent = 4, sort_keys = True)
+                else:
+                    for dataItem in data[key]:
+                        f.write("\t".join([str(d) for d in dataItem]) + "\n")
+
+    def __prepareDataToWrite(self, knowsMatrix, people):
+        """
+        Prepare data for writing to file.
+        """
+        n = knowsMatrix.shape[1]
+
+        knowsData = []
+        knowsObs = []
+        knowsTarget = []
+        knowsTruth = []
+        
+        totalCombinations = knowsMatrix.shape[0] * (knowsMatrix.shape[1] - 1)
+        targets = np.random.choice(range(totalCombinations), 
+                                    int(self.targetSplit * totalCombinations), replace = False)
+        for i in range(knowsMatrix.shape[0]):
+            for j in range(knowsMatrix.shape[1]):
+                if i != j:
+                    knowsData.append([i, j, knowsMatrix[i][j]])
+                if (n * i + j) not in targets and i != j:
+                    knowsObs.append([i, j, knowsMatrix[i][j]])
+                elif (n * i + j) in targets and i != j:
+                    knowsTarget.append([i, j])
+                if (n * i + j) in targets and i != j:
+                    knowsTruth.append([i, j, knowsMatrix[i][j]])
+
+        livedObs = []
+        for person in people:
+            for index, place in enumerate(person.lived):
+                if place == 1:
+                    livedObs.append([person.index, index])
+
+        likesObs = []
+        for person in people:
+            for globalLike in person.globalLikes:
+                likesObs.append([person.index, globalLike, person.globalLikes[globalLike]])
+
+        # NOTE: Both global and local likes are combined in a single list called likesObs.
```

Comment:
Don't use "NOTE" like a tag (e.g. "TODO" or "TEST") here.
Since a comment is already a form of a note, just use it more naturally like:
`# Note that both global ...`